### PR TITLE
Revise AWS terraform and docs

### PR DESCRIPTION
**Pull request by joshk0**
* Use consistent role name across CFN and TF
* All TF files become modules that can be used in a `module` block
* Add better readmes explaining why you need to use each script (or not)

### CODE REVIEW

File: aws/centralized-monitoring/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+If you are using the [aws-centralized-monitoring] script and are directing
+metrics to flow into the account where you applied the [payer-account] role,
+then you do not need to apply this. Simply use the same role ARN generated from
+the `payer-account` module.
+
+## Example terraform usage
+
+```hcl
+module "ternary-centralized-monitoring" {
+    source = "https://github.com/TernaryInc/ternary-onb-permissions.git//aws/centralized-monitoring?ref=master"
```
Comment (asproul) [inline aws/centralized-monitoring/README.md:24]: Did this work for you?  

I got it to work with
```
source = "github.com/TernaryInc/ternary-onb-permissions//aws/centralized-monitoring?ref=docz"
```

File: aws/payer-account/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+To obtain the value for `ternary_bucket_id`, follow the instructions to
+set up an AWS CUR export to an S3 bucket. This bucket ID is what you should
+use for the value.
+
+## Example terraform usage
+
+```hcl
+module "ternary-payer-account" {
+    source = "https://github.com/TernaryInc/ternary-onb-permissions.git//aws/payer-account?ref=master"
```
Comment (asproul) [inline aws/payer-account/README.md:22]: Were you able to get this one to work? 

I was able to use

```
source = "github.com/TernaryInc/ternary-onb-permissions//aws/payer-account?ref=docz"
```

File: aws/centralized-monitoring/variables.tf [hunk 3]
```diff
@@ -0,0 +1,9 @@
+variable "ternary_service_account_email" {
+  type        = string
+  description = "Service Account email used by your Ternary tenant to access your cloud resources. Available in your Admin page."
+}
+
+variable "ternary_service_account_uid" {
+  type        = string
+  description = "Service Account Unique ID used by your Ternary tenant to access your cloud resources. Available in your Admin page."
+}
```
Comment (asproul) [inline aws/centralized-monitoring/variables.tf:9]: What do you think about adding a [variable validation](https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules) to ensure the variables conform to the expected values?

Maybe something like this
```suggestion
variable "ternary_service_account_uid" {
  type        = string
  description = "Service Account Unique ID used by your Ternary tenant to access your cloud resources. Available in your Admin page."
  validation {
    condition     = length(regexall("[0-9]+", var.ternary_service_account_uid)) > 0
    error_message = "The ternary_service_account_uid value must be numeric."
  }
}
```

**Review by asproul — COMMENTED** [review]
LGTM. Left a couple of comments.

File: aws/centralized-monitoring/README.md (hunk 1 above)
Comment (joshk0) [reply aws/centralized-monitoring/README.md:24]: It will be `master` once merged, right?

Comment (asproul) [reply aws/centralized-monitoring/README.md:24]: yea.. was testing with this branch at the time but, yes it should be "master" once this is merged.

File: aws/linked-account/variables.tf [hunk 4]
```diff
@@ -0,0 +1,9 @@
+variable "ternary_service_account_email" {
+  type        = string
+  description = "Service Account email used by your Ternary tenant to access your cloud resources. Available in your Admin page."
+}
+
+variable "ternary_service_account_uid" {
+  type        = string
+  description = "Service Account Unique ID used by your Ternary tenant to access your cloud resources. Available in your Admin page."
+}
```
Comment (asproul) [inline aws/linked-account/variables.tf:9]:
```suggestion
variable "ternary_service_account_uid" {
  type        = string
  description = "Service Account Unique ID used by your Ternary tenant to access your cloud resources. Available in your Admin page."
  validation {
    condition     = length(regexall("[0-9]+", var.ternary_service_account_uid)) > 0
    error_message = "The ternary_service_account_uid value must be numeric."
  }
}}
```

File: aws/payer-account/variables.tf [hunk 5]
```diff
@@ -0,0 +1,14 @@
+variable "ternary_service_account_email" {
+  type        = string
+  description = "Service Account email used by your Ternary tenant to access your cloud resources. Available in your Admin page."
+}
+
+variable "ternary_service_account_uid" {
+  type        = string
+  description = "Service Account Unique ID used by your Ternary tenant to access your cloud resources. Available in your Admin page."
+}
```
Comment (asproul) [inline aws/payer-account/variables.tf:9]:
```suggestion
variable "ternary_service_account_uid" {
  type        = string
  description = "Service Account Unique ID used by your Ternary tenant to access your cloud resources. Available in your Admin page."
  validation {
    condition     = length(regexall("[0-9]+", var.ternary_service_account_uid)) > 0
    error_message = "The ternary_service_account_uid value must be numeric."
  }
}
```

**Review by asproul — APPROVED** [review]

**Pull request merged**